src/components/home/Home.tsx

```diff
@@ -1,10 +1,60 @@
 import { ReactElement } from "react";
+import GitHub from "../icons/gith-hub";
+import { Button } from "../ui/button";
+import { Link } from "react-router-dom";
+import * as qs from 'qs';
+import { generateCSRFToken } from "../../lib/security";
+import { useStoreInContext } from "../../lib/zustand";
+import { ProviderName } from "../../requests/integrations/oauth2/github";
+
+// TODO[P3]: Support for other git providers
+const ACCEPTED_GIT_PROVIDERS = new Set(["github"]);
+
+const LoginWithGithubButton = () => {
+    // TODO: store this in local storage & verify against it on callback
+    const csrfToken = generateCSRFToken();
+
+    const queryString = qs.stringify({
+        client_id: import.meta.env.VITE_GITHUB_OAUTH_APP_CLIENT_ID,
+        redirect_uri: import.meta.env.VITE_GITHUB_OAUTH_APP_REDIRECT_URI,
+        response_type: "code",
+        scope: "repo",
+        state: csrfToken
+    });
+
+    return (
+        <Link to={`https://github.com/login/oauth/authorize?${queryString}`} >
+            <Button className={"bg-black dark:bg-white text-xl space-x-2"} >
+                <GitHub height={4} width={4} />
+                <p>Login with GitHub</p>
```

Comment:
Use of `<p></p>`  tag is not correct/necessary. See [this link](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button).

Better to only add elements when needed. For example, you have a section of the text you want styled red, it would look like:

```html
<Button>
    Text goes <span className="text-red-500">here</span>
</ Button>
```